src/client.ts

```diff
@@ -424,4 +461,35 @@ export class AutoblocksAPIClient {
       `/datasets/${encodedName}/schema-versions/${encodedSchemaVersion}${splitsQueryParam}`,
     );
   }
+
+  public async createDatasetItem(args: {
+    datasetExternalId: string;
+    data: Record<string, unknown>;
+  }): Promise<{ id: string }> {
+    return this.post(
+      `/datasets/${encodeURIComponent(args.datasetExternalId)}/items`,
+      { data: args.data },
+    );
+  }
+
+  public async deleteDatasetItem(args: {
+    datasetExternalId: string;
+    itemId: string;
+  }): Promise<{ id: string }> {
+    return this.delete(
+      `/datasets/${encodeURIComponent(args.datasetExternalId)}/items/${encodeURIComponent(args.itemId)}`,
+    );
+  }
+
+  public async updateDatasetItem(args: {
+    datasetExternalId: string;
```

Comment:
lets call this name so its consistent with getDataset. externalId is really an internal naming convention